File: old_backend/important-core-code/java/VulnerableCommand.java
// INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
// This file contains critical command injection vulnerabilities for testing Aikido scanner

import java.io.*;

public class VulnerableCommand {
    
    public void pingHost(String host) throws IOException {
        // VULNERABLE: Runtime.exec with concatenation
        Runtime rt = Runtime.getRuntime();
        Process proc = rt.exec("ping -c 4 " + host);
    }
    
    public void processFile(String filename) throws IOException {
        // VULNERABLE: ProcessBuilder with user input
        ProcessBuilder pb = new ProcessBuilder("convert", filename, "output.png");
        pb.start();
    }
    
    public void executeCommand(String cmd) throws IOException {
        // VULNERABLE: Shell command execution
        String[] command = {"/bin/sh", "-c", cmd};
        Runtime.getRuntime().exec(command);
    }
}

File: old_backend/important-core-code/ruby/vulnerable_command.rb
# INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
# This file contains critical command injection vulnerabilities for testing Aikido scanner

class SystemController < ApplicationController
  def ping
    host = params[:host]
    
    # VULNERABLE: Backticks with user input
    output = `ping -c 4 #{host}`
    render plain: output
  end
  
  def process
    filename = params[:file]
    
    # VULNERABLE: system() with interpolation
    system("convert #{filename} output.png")
  end
  
  def execute
    command = params[:cmd]
    
    # VULNERABLE: exec with user input
    exec("ls -la #{command}")
  end
end
